BootCampV2/Homeworks/SecondWeek/ArrayExtension.cs

```diff
@@ -0,0 +1,149 @@
+using System;
+
+namespace BootCampV2
+{
+    public class ArrayExtension
+    {
+        /// <summary>
+        /// Use the bubble sort algorithm to sort an array
+        /// <para> return: sorted Array </para> 
+        /// </summary>
+        public static int[] BubbleSort(int[] array)
+        {
+            for (int i = 0; i < array.Length; i++)
+            {
+                bool swap = false;
+                for (int m = 0; m < (array.Length - i) - 1; m++)
+                {
+                    if (array[m] > array[m + 1])
+                    {
+                        array = SwapElements(array, m, m + 1);
+                        swap = true;
+                    }
+                }
+                if (!swap)
+                    break;
+            }
+            return array;
+        }
+
+
+        /// <summary>
+        /// Swap two elements with each other
+        /// <para> return: modified Array </para> 
+        /// </summary>
+        public static int[] SwapElements(int[] array, int firstIndex, int secondIndex)
+        {
+            if (firstIndex >= array.Length || firstIndex < 0 ||
```

Comment:
Normally I might think SwapElements to be a private helper method, then you wouldn't have to do validation (if you pass bad input to a helper method you can generally just let it throw an exception, input validation should happen when input comes from external sources) and you wouldn't have to comment it (typically you don't comment private helper methods but only methods you are making for other people to use).  By making it public you are making more work for yourself.  It isn't a bad thing necessarily but something to consider.